### PR TITLE
chore(lint): use golangci-lint to call revive and misspell checker.

### PR DESCRIPTION
**Pull request by appleboy**
replace revive and misspell with golangci-lint

Signed-off-by: Bo-Yi Wu <appleboy.tw@gmail.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -280,29 +278,6 @@ errcheck:
 	@echo "Running errcheck..."
 	@errcheck $(GO_PACKAGES)
 
-.PHONY: revive
-revive:
```
Comment (Gusted) [inline Makefile:284]: I could see why we would remove revive, I'm not sure if it actually could be a drop-in as I know from golangci-lint that they don't always have 100% configuration as if you have with cmd command.

File: Makefile [hunk 2]
```diff
@@ -280,29 +278,6 @@ errcheck:
...
-	fi
-	@revive -config .revive.toml -exclude=./vendor/... ./...
-
-.PHONY: misspell-check
-misspell-check:
-	@hash misspell > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
-		$(GO) install github.com/client9/misspell/cmd/misspell@v0.3.4; \
-	fi
-	@echo "Running misspell-check..."
-	@$(GO) run build/code-batch-process.go misspell -error -i unknwon '{file-list}'
```
Comment (Gusted) [inline Makefile:296]: It seems to me that this shouldn't be touched, as it's using `build/code-batch-process.go` which seems to be special build made to batch this misspell action. Not sure why, maybe one of the other maintainers can shed some light on that.

Comment (wxiaoguang) [reply Makefile:296]: `build/code-batch-process.go` is only a wrapper for CLI.

In Windows, you can not pass more than 32K bytes via CLI arguments, so the`code-batch-process.go` split the file names into batches. We can just ignore the wrapper during refacotoring.

Comment (wxiaoguang) [reply Makefile:296]: ps: the `build/code-batch-process.go` came from https://github.com/go-gitea/gitea/pull/17684 to fix https://github.com/go-gitea/gitea/issues/14438 , so do not worry about it 😊

File: Makefile [hunk 3]
```diff
@@ -157,7 +157,7 @@ all: build
 .PHONY: help
 help:
 	@echo "Make Routines:"
-	@echo " - \"\"                               equivalent to \"build\""
```
Comment (wxiaoguang) [inline Makefile]: We shouldn't remove this tab. If we remove it, the output might be broken.

The outputted texts were already aligned before:

image

**Comment by wxiaoguang** [comment]
For my understanding, this PR just removes `revive` and `misspell`.

The comment said `replace revive and misspell with golangci-lint`. How are they replaced?

**Comment by lunny** [comment]
`golangci-lint` now use `revive` internally #17609 but I think `misspell` is still necessary.

File: Makefile (hunk 3 above)
Comment (wxiaoguang) [reply Makefile]: We could use `'` instead of `"` for this line to make the source code more beautiful.

File: Makefile (hunk 1 above)
Comment (appleboy) [reply Makefile:284]: maybe we should integrate revive in golangci-lint?

File: Makefile (hunk 3 above)
Comment (appleboy) [reply Makefile]: @wxiaoguang reverted this change.

**Comment by appleboy** [comment]
@wxiaoguang we can enable the `revive` lint in golangci-lint, see https://github.com/go-gitea/gitea/pull/18145/commits/c96f3ce58360d072b753327e66c90dad635322b5 and `misspell` has been added in yaml file.

**Comment by lunny** [comment]
> @wxiaoguang we can enable the `revive` lint in golangci-lint, see [c96f3ce](https://github.com/go-gitea/gitea/commit/c96f3ce58360d072b753327e66c90dad635322b5) and `misspell` has been added in yaml file.

Which one is the misspell check?

**Comment by appleboy** [comment]
@lunny See https://github.com/go-gitea/gitea/blob/549fd03c0e9aa19850c8271cbd7cd62bbc884b34/.golangci.yml#L15

**Review by lunny — APPROVED** [review]

**Review by Gusted — APPROVED** [review]
Faster linting 🎉

**Comment by zeripath** [comment]
make lgtm work

**Comment by wxiaoguang** [comment]
The document should also be update, eg: https://docs.gitea.io/en-us/hacking-on-gitea/#formatting-code-analysis-and-spell-check : `make revive vet misspell-check`

**Comment by wxiaoguang** [comment]
--> Remove code-batch-process #21471

**Pull request merged**